### ISSUE TITLE
[Bug] Waypoints reduce in size with distance as if they're objects, and scale modifier doesn't help much

### ISSUE THREAD

**Issue by w014prc2** [MeteorDevelopment/meteor-client#1474]
### Describe the bug

tl;dr
waypoints get way too small (and basically disappear) when you get too far away from them, and the only way to see them is to put obscenely large numbers into the scale modifier.

I wouldn't usually say something like this is a bug, its more just a poor choice of mechanics, but it is such a hassle that, if the problem was known about before, it wouldn't have been allowed to make it in to the release (either that, or I don't use waypoints how they're meant to be used).

I often use waypoints to locate points that are very far away (I use it to walk in the exact direction of a base that could be millions of blocks away). This is fine when waypoints is working properly; however, with the newest round of updates (which has admittedly been around for a while) waypoints was modified.

Previously, there was a bug where, the farther away the player was from a given waypoint, the larger the waypoint appeared. After that bug got fixed, it was nice that waypoints weren't taking up half the screen, but the fix fixed it _too_ much, if you know what I mean. Now, a waypoint is the size of a block, and as such, if you walk only a few hundred blocks away, you can't see it anymore! Even worse, the scale option only scales the _size_ of the waypoint _linearly!_ Because of this, when I am travelling to far away bases, I can't set a waypoint like I used to. Sometimes it works to add a bunch of 0s to the scale modifier, but only if you are close enough. I am currently 100k away from a base, and have both scale and distance modifiers full of 0s (I checked to make sure they both have 1s at the beginning, and I double checked that waypoints was turned on), yet I can't see any waypoints. It is also infuriating when I have a bunch of waypoints close together, and I am either too far away to read which one is which, or when I approach one, it is massive.

Basically, all I am asking is, make the scale not linear, but like an inverse square: something so that at a certain distance, the size of the waypoint stops getting smaller.

### Steps to reproduce

```bash
1. Make a waypoint (do it in creative, walking that far in survival would take too long)
2. Teleport an arbitrary (ideally large) distance away
3. See how large of a value you need to input into the scale modifier to make the waypoint visible (a lot)
```

### Meteor Version

0.4.5

### Minecraft Version

1.17.1

### Operating System

Windows

**Comment by cum-account** [comment]
or add a min/max waypoint size slider

**Comment by seasnail8169** [comment]
duplicate #1150
